esp/bindings/http/platform/httpservice.cpp

```diff
@@ -351,6 +397,46 @@ int CEspHttpServer::processRequest()
         m_request->getEspPathInfo(stype, &pathEx, &serviceName, &methodName, false);
         ESPLOG(LogNormal,"sub service type: %s. parm: %s", getSubServiceDesc(stype), m_request->queryParamStr());
 
+        // getEspPathInfo provides all information needed to decide if the request should be
+        // traced. Create a server span, if needed, before proceding with request processing
+        // so maximize the amount of request processing that can be traced. Specifically, user
+        // authentication and authorization may generate trace output.
+        bool wantTracing = true;
+        UntracedGetMethodMap::const_iterator untracedRequestIt = getRequests.end();
+        if (!queryTraceManager().isTracingEnabled())
+            wantTracing = false;
+        else if (streq(method, GET_METHOD))
+        {
+            if (sub_serv_root == stype)
+                wantTracing = false;
+            else if (strieq(serviceName, "esp"))
+            {
+                if (methodName.isEmpty())
+                    wantTracing = false;
+                else
+                {
+                    // The presence of a method name in the get request map is sufficient to
+                    // suppress trace output. The enumerated value will be used later.
+                    untracedRequestIt = getRequests.find(methodName);
```

Comment:
This will not be executed if !isTracingEnabled, which means the switch statement later on will not work.  This needs to be unconditional - and because it is now using a map it should be ok to be unconditional.